pymeasure/instruments/anritsu/anritsuMS464xB.py

```diff
@@ -0,0 +1,844 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+import logging
+
+from pymeasure.instruments import Instrument, Channel
+from pymeasure.instruments.validators import (
+    strict_discrete_set,
+    strict_range
+)
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class AnritsuMS464xB(Instrument):
+    """ A class representing the Anritsu MS464xB Vector Network Analyzer (VNA) series. Supports the
+    MS4642B, MS4644B, MS4645B, and MS4647B, which are represented in their respective classes
+    (:class:`~.AnritsuMS4642B`, :class:`~.AnritsuMS4644B`, :class:`~.AnritsuMS4645B`,
+    :class:`~.AnritsuMS4647B`), that only differ in the available frequency range.
+
+    Can contain up to 16 instances of :class:`~.MeasurementChannel` (depending on the configuration
+    of the instrument), that are accessible via the `channels` dict or directly via `ch_` + the
+    channel number.
+    """
+    CHANNELS = [1, 16]
+    TRACES = [1, 16]
+    PORTS = [1, 4]
+    TRIGGER_TYPES = ["POIN", "SWE", "CHAN", "ALL"]
+
+    FREQUENCY_RANGE = [1E7, 7E10]
+
+    SPARAM_LIST = ["S11", "S12", "S21", "S22",
+                   "S13", "S23", "S33", "S31",
+                   "S32", "S14", "S24", "S34",
+                   "S41", "S42", "S43", "S44", ]
+    DISPLAY_LAYOUTS = ["R1C1", "R1C2", "R2C1", "R1C3", "R3C1",
+                       "R2C2C1", "R2C1C2", "C2R2R1", "C2R1R2",
+                       "R1C4", "R4C1", "R2C2", "R2C3", "R3C2",
+                       "R2C4", "R4C2", "R3C3", "R5C2", "R2C5",
+                       "R4C3", "R3C4", "R4C4"]
+
+    def __init__(self, adapter, name="Anritsu MS464xB Vector Network Analyzer", **kwargs):
+        super().__init__(
+            adapter,
+            name,
+            timeout=10000,
+            **kwargs,
+        )
+
+        self.update_channels(self.CHANNELS[1])
```

Comment:
IIUC, the number of channels can be (have to be) chosen at runtime because of the way the specific instrument is connected/equipped with hardware modules or somesuch? In that case, maybe add a `channels` argument to the init signature, and use that?